graylog-storage-elasticsearch7/src/test/java/org/graylog/storage/elasticsearch7/testing/ClientES7.java

```diff
@@ -90,22 +96,55 @@ public void addAliasMapping(String indexName, String alias) {
                 .alias(alias);
         indicesAliasesRequest.addAliasAction(aliasAction);
 
-        client.execute((c, requestOptions) -> c.indices().updateAliases(indicesAliasesRequest, requestOptions));
+        client.execute((c, requestOptions) -> c.indices().updateAliases(indicesAliasesRequest, requestOptions),
+                "failed to add alias " + alias + " for index " + indexName);
     }
 
     @Override
-    public JsonNode getMapping(String... indices) {
-        return null;
+    public boolean isSourceEnabled(String testIndexName) {
+        // TODO: implement
+        return true;
     }
 
     @Override
-    public JsonNode getTemplate(String templateName) {
-        return null;
+    public String fieldType(String testIndexName, String field) {
+        return getMapping(testIndexName).get(field);
+    }
+
+    private Map<String, String> getMapping(String index) {
+        final Request request = new Request("GET", "/" + index + "/_mapping");
+        final Response result = client.execute((c, requestOptions) -> {
+            request.setOptions(requestOptions);
+            return c.getLowLevelClient().performRequest(request);
+        });
+
+        final JsonNode response;
+        try {
+            response = objectMapper.readTree(result.getEntity().getContent());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        return Streams.stream(response.path(index)
+                .path("mappings")
+                .path("properties")
+                .fields())
+                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().path("type").asText()));
```

Comment:
I'd suggest extracting a private variable here. (I spotted the closing bracket for `Streams#stream` on line 130 by accident and it just bugged me...) 
```suggestion
        final Iterator<Map.Entry<String, JsonNode>> fields = response.path(index)
                .path("mappings")
                .path("properties")
                .fields();
        //noinspection UnstableApiUsage
        return Streams.stream(fields)
                .collect(Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().path("type").asText()));
```

Reply:
:heavy_check_mark: I extracted a helper method for it.

Reply:
Even better!